Source/core/FileSystem.cpp

```diff
@@ -56,6 +56,90 @@ namespace Core {
     {
         Close();
     }
+
+    /* static */ string File::Normalize(const string& location, bool& valid)
+    {
+        string result(location);
+
+        valid = true;
+
+        // First see if we are not empy.
+        if (result.empty() == false) {
+            uint32_t index = 0;
+
+           while (index < result.length()) {
+
+#ifdef __WINDOWS__
+                if (result[index] == '\\') {
+                    result[index] = '/';
+                }
+#endif
+
+                if ((result[index] == '/') && (index >= 1) ) {
+                    if (result[index - 1] == '/') {
+                        if (index >= 2) {
+                            // We have a double slash, clear all till the beginning
+                           result = result.substr(index - 1);
+                           index = 1;
```

Comment:
You almost got met here, I thought shouldn't it be 2? But indeed we now skip to the end and do ++ and start the next iteration at 2 :)

Reply:
oke.